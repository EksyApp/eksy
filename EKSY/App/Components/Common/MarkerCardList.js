import React, {Component} from 'react'
import {ListView, Dimensions} from 'react-native'
import MarkerCard from '../MainView/BottomSheet/MarkerCard'
import PropTypes from 'prop-types';
import {MarkersShape} from "../../Utils/PropTypeShapes";

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

export default class MarkerCardList extends Component {

  constructor(props) {
		super(props)

		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1 !== r2})

		this.state = {
			dataSource: datasource.cloneWithRows(this.props.data)
		}

	}

  componentWillReceiveProps(props) {
		if(props.data !== this.props.data) {
			const dataSource = this.state.dataSource.cloneWithRows(props.data || []);
			this.setState({dataSource})
		}
	}

  render() {
    return(
      <ListView
        dataSource={this.state.dataSource}
        renderRow = {(marker) => {
          return(
            <MarkerCard
            width={Screen.width-10}
            marker={marker}
            onPress={this.props.onPress}
            style={this.props.style}
          />
        )
        }}
        enableEmptySections={true}
      />

    )
  }
}

MarkerCardList.PropTypes = {
	data: MarkersShape,
	onPress: PropTypes.func,
	style: PropTypes.object
}