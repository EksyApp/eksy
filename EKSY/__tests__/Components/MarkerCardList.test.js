import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MarkerCardList from "../../App/Components/Common/MarkerCardList";

test('MarkerCardList renders correctly', () => {
	const tree = renderer.create(
			<MarkerCardList
					data={[]}
					onPress={() => {}}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});