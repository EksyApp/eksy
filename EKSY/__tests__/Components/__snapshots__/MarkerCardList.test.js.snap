// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerCardList renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  enableEmptySections={true}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View />
</RCTScrollView>
`;
